tldw_Server_API/app/core/DB_Management/ChaChaNotes_DB.py

```
@@ -5686,6 +5686,34 @@ def get_messages_for_conversation(self, conversation_id: str, limit: int = 100,
+        """Check whether a conversation has at least one system message."""
+        delete_clause = "" if include_deleted else "AND m.deleted = 0"
+        query = f"""
+            SELECT 1
+            FROM messages m
+            JOIN conversations c ON m.conversation_id = c.id
+            WHERE m.conversation_id = ?
+              AND lower(m.sender) = 'system'
+              {delete_clause}
+              AND c.deleted = 0
+            LIMIT 1
+        """
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)